components/server/src/session-handler.ts

```diff
@@ -39,7 +42,31 @@ export class SessionHandlerProvider {
 
         options.store = this.createStore();
 
-        this.sessionHandler = session(options);
+        this.sessionHandler = (req, res, next) => {
+            let hasJWTCookie = false;
+
+            const cookies = parseCookieHeader(req.headers.cookie || "");
```

Comment:
For some reason, `req.cookies` always contains the empty object, and never any of the actual cookies in the request. This is despite us using the `express-cookie` middleware which should populate it.

Doesn't matter much, as getting the token out of the header is simple enough, but it is odd.

Reply:
Jup, this never worked. I think we are missing to set the first parameter `secret` of `cookieParser()` [here](https://github.com/gitpod-io/gitpod/blob/bcca71039ca01f0d833b487d036b1b03cb4c3335/components/server/src/server.ts#L155), to the same value we use [here](https://github.com/gitpod-io/gitpod/blob/233ec9457a99d935a8983e4229409ed652a76b2d/components/server/src/session-handler.ts#L37). 

Maybe worth playing around with.

Reply:
I'll investigate in another PR.